generators/CuraPackage.py

```diff
@@ -0,0 +1,177 @@
+import os
+import shutil
+import json
+import sysconfig
+
+from pathlib import Path
+
+from jinja2 import Template
+
+from conans.errors import ConanInvalidConfiguration
+from conans.model import Generator
+from conans.util.files import mkdir, sha256sum
+from conans.tools import Version
+
+FILTERED_FILES = ["conaninfo.txt", "conanmanifest.txt"]
+FILTERED_DIRS = ["include"]
+
+
+class CuraPackage(Generator):
+    _content_types_xml = r"""<?xml version="1.0" encoding="UTF-8"?>
+<Types xmlns="http://schemas.openxmlformats.org/package/2006/content-types">
+  <Default ContentType="application/vnd.openxmlformats-package.relationships+xml" Extension="rels" />
+  <Default ContentType="application/x-ultimaker-material-profile" Extension="xml.fdm_material" />
+  <Default ContentType="application/x-ultimaker-material-sig" Extension="xml.fdm_material.sig" />
+  <Default ContentType="application/x-ultimaker-quality-profile" Extension="inst.cfg" />
+  <Default ContentType="application/x-ultimaker-machine-definition" Extension="def.json" />
+  <Default ContentType="text/json" Extension="json" />
+</Types>"""
+
+    _dot_rels = r"""<?xml version="1.0" encoding="UTF-8"?>
+<Relationships xmlns="http://schemas.openxmlformats.org/package/2006/relationships">
+  <Relationship Target="/package.json" Type="http://schemas.ultimaker.org/package/2018/relationships/opc_metadata" Id="rel0" />
+</Relationships>"""
+
+    _package_json_rels = r"""<?xml version="1.0" encoding="UTF-8"?>
+<Relationships xmlns="http://schemas.openxmlformats.org/package/2006/relationships">
+  <Relationship Target="/plugins" Type="plugin" Id="rel0" />
+</Relationships>"""
+
+    _deps_init_py = Template(r"""def initialize_paths():
+    {% if py_deps | length > 0 or bin_deps | length > 0 %}from pathlib import Path
+    {% if bin_deps | length > 0 %}from platform import system{% endif %}
+    {% if py_deps | length > 0 %}from sys import path
+    import sysconfig
+
+    platform_path = f"python{sysconfig.get_config_var('VERSION')}_{sysconfig.get_platform().replace('-', '_')}"{% endif %}  
+    {% if py_deps | length > 0 %}{% for dep in py_deps %}path.append(str(Path(__file__).parent.parent.joinpath(r"{{ dep }}", platform_path))){% endfor %}{% endif %}
+
+    {% if bin_deps | length > 0 %}if system() == "Windows":
+        from os import add_dll_directory
+        {% for dep in bin_deps %}add_dll_directory(str(Path(__file__).parent.joinpath(r"{{ dep }}")))
+        {% endfor %}{% endif %}{%  else %}pass{% endif %}
+
+initialize_paths()
+
+""")
+
+    def subdict(self, base, *keys, **kwargs):
+        sub = dict(zip(keys, [base[k] for k in keys]))
+        for k, v in kwargs.items():
+            sub[k] = base[v]
```

Comment:
I don't mind the way this is done here, it's unlikely to cause any error (since the caller has to know about the extra optional arguments in the first place). It can also not really cause a slowdown, since it won't iterate over something that hasn't been given.